src/http-client.js

```diff
@@ -105,12 +105,13 @@ export class HttpClient {
         if (Response.prototype.isPrototypeOf(result)) {
           response = result;
         } else if (Request.prototype.isPrototypeOf(result)) {
+          request = Promise.resolve(result);
           response = fetch(result);
         } else {
           throw new Error(`An invalid result was returned by the interceptor chain. Expected a Request or Response instance, but got [${result}]`);
         }
 
-        return processResponse(response, this.interceptors);
```

Comment:
Oh yes, good point. Actually, this is probably an issue in the existing
code. I don't think it bails out of applying the request interceptors if
the first one returns a Response.

On Thu, Jan 21, 2016 at 11:34 PM timbell <notifications@github.com> wrote:

> In src/http-client.js
> <https://github.com/aurelia/fetch-client/pull/34#discussion_r50508402>:
>
> >            response = fetch(result);
> >          } else {
> >            throw new Error(`An invalid result was returned by the interceptor chain. Expected a Request or Response instance, but got [${result}]`);
> >          }
> >
> > -        return processResponse(response, this.interceptors);
>
> But can't result be a Response if request or requestError return a
> Response directly and we'd then be passing a Response as a Request to
> response/responseError?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aurelia/fetch-client/pull/34/files#r50508402>.
>